src/main/kotlin/com/cultureamp/eventsourcing/PGSessionAdvisoryBookmarkLock.kt

```diff
@@ -0,0 +1,70 @@
+package com.cultureamp.eventsourcing
+
+import org.slf4j.LoggerFactory
+import java.sql.Connection
+import java.sql.SQLException
+import java.util.UUID
+
+private val logger = LoggerFactory.getLogger(PGSessionAdvisoryBookmarkLock::class.java)
+class PGSessionAdvisoryBookmarkLock(
+    private val lockTimeoutMs: Int = 60_000,
+    private val connectionProvider: () -> Connection,
+) {
+    private val lock = Any()
+    private var connection = getConnection()
+
+    fun lock(bookmarkName: String) {
+        synchronized(lock) {
+            try {
+                refreshLock(connection, bookmarkName)
+            } catch (e: Exception) {
+                logger.info("Failed to refresh session lock for $bookmarkName, resetting connection to try again", e)
+                connection.close()
+                connection = getConnection()
```

Comment:
This implies a new connection every time `lock` is called

Reply:
Why's that? it only does this if an exception is thrown, and the exception doesn't meet [this criteria](https://github.com/cultureamp/kestrel/pull/62/files/ddb646b35debbb099445c1fc3032f99267b8385b#diff-8fe24a5dc3b2a10915166c71bee3a26ae20a8f3cea7ced1505e431d9326315dcR57)